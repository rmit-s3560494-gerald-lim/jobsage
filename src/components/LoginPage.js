import React, { Component } from "react";
import './App.css';
import axios from 'axios';

class LoginPage extends Component {

  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: '',
    };
  }

  handleChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    });
  }

  validateUser = () => {
    axios.post('http://35.212.88.235/users/validation/', {
      email: this.state.email,
      password: this.state.password,
    }).then(response => {
        if(response.data[0].user_type === 'employer'){
          localStorage.setItem('isEmployerLoggedIn', 'true');
          localStorage.setItem('user', JSON.stringify(response.data[0]));
          this.console.log("test");
          this.props.history.push('/employerhome');
        } else {
          localStorage.setItem('isEmployeeLoggedIn', 'true');
          localStorage.setItem('user', JSON.stringify(response.data[0]));
          this.props.history.push('/employeehome');
        }
    }).catch(error => {
      console.log('Error: ', error);
    })
  }

  handleSubmit = (e) => {
    e.preventDefault();
    this.validateUser();
  }

  render() {
    return (
      <div>
        <div class="centerLogin">
        <div class="form">

          <form onSubmit={this.handleSubmit}>
            <div class="form-group">
              <label for="emailLabel">Email address</label>
              <input type="email" class="form-control" name="email" onChange={this.handleChange} id="email" aria-describedby="emailHelp" placeholder="Enter email" />
              <small id="emailHelp" class="form-text text-muted">We'll never share your email with anyone else.</small>
            </div>
            <div class="form-group">
              <label for="passwordLabel">Password</label>
              <input type="password" class="form-control" name="password" onChange={this.handleChange} id="exampleInputPassword1" placeholder="Password" />
            </div>
            <button type="submit" class="btn btn-primary">Submit</button>
          </form>
          <a href="/signup">Don't have an account?</a>
          </div>
        </div>
      </div>
    );
  }
}

export default LoginPage;
