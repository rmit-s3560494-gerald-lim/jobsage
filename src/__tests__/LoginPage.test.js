import React from 'react';
import { shallow, mount } from 'enzyme';
import LoginPage from '../components/LoginPage';

describe('LoginPage Unit Tests', () => {
  test('renders', () => {
    const wrapper = shallow(<LoginPage />);
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });


  it('renders the logo', () => {
    const wrapper = shallow(<LoginPage />);
    expect(wrapper.html()).toContain('img src="logo.png"');
  });

  xit('has an input for email', () => {
    const wrapper = mount(<LoginPage />);
    const input = wrapper.find('input');
    expect(input.props('name')).toEqual('email');
  })

  it('has a submit button', () => {
    const wrapper = mount(<LoginPage />);
    expect(wrapper.containsMatchingElement(
      <button type="submit" className="btn btn-primary">Submit</button>
    )).toBeTruthy();
  })
});