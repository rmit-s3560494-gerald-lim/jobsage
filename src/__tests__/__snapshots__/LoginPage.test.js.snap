// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage Unit Tests renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="centerLogin"
      >
        <img
          alt="JobSage Logo"
          className="img-fluid"
          src="logo.png"
        />
        <h1
          className="display-3"
          id="applicationTitle"
        >
          JobSage
        </h1>
        <div
          className="form"
        >
          <form
            className="loginform"
            onSubmit={[Function]}
          >
            <div
              className="form-group"
            >
              <label>
                Email Address
              </label>
              <input
                aria-describedby="emailHelp"
                className="form-control"
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Enter email"
                type="email"
              />
            </div>
            <div
              className="form-group"
            >
              <label>
                Password
              </label>
              <input
                className="form-control"
                id="exampleInputPassword1"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
              />
            </div>
            <button
              className="btn btn-primary"
              type="submit"
            >
              Submit
            </button>
          </form>
          <a
            href="/signup"
            id="signUpLink"
          >
            Don't have an account?
          </a>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="JobSage Logo"
            className="img-fluid"
            src="logo.png"
          />,
          <h1
            className="display-3"
            id="applicationTitle"
          >
            JobSage
          </h1>,
          false,
          <div
            className="form"
          >
            <form
              className="loginform"
              onSubmit={[Function]}
            >
              <div
                className="form-group"
              >
                <label>
                  Email Address
                </label>
                <input
                  aria-describedby="emailHelp"
                  className="form-control"
                  id="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="Enter email"
                  type="email"
                />
              </div>
              <div
                className="form-group"
              >
                <label>
                  Password
                </label>
                <input
                  className="form-control"
                  id="exampleInputPassword1"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                />
              </div>
              <button
                className="btn btn-primary"
                type="submit"
              >
                Submit
              </button>
            </form>
            <a
              href="/signup"
              id="signUpLink"
            >
              Don't have an account?
            </a>
          </div>,
        ],
        "className": "centerLogin",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "JobSage Logo",
            "className": "img-fluid",
            "src": "logo.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "JobSage",
            "className": "display-3",
            "id": "applicationTitle",
          },
          "ref": null,
          "rendered": "JobSage",
          "type": "h1",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <form
                className="loginform"
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <label>
                    Email Address
                  </label>
                  <input
                    aria-describedby="emailHelp"
                    className="form-control"
                    id="email"
                    name="email"
                    onChange={[Function]}
                    placeholder="Enter email"
                    type="email"
                  />
                </div>
                <div
                  className="form-group"
                >
                  <label>
                    Password
                  </label>
                  <input
                    className="form-control"
                    id="exampleInputPassword1"
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                  />
                </div>
                <button
                  className="btn btn-primary"
                  type="submit"
                >
                  Submit
                </button>
              </form>,
              <a
                href="/signup"
                id="signUpLink"
              >
                Don't have an account?
              </a>,
            ],
            "className": "form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="form-group"
                  >
                    <label>
                      Email Address
                    </label>
                    <input
                      aria-describedby="emailHelp"
                      className="form-control"
                      id="email"
                      name="email"
                      onChange={[Function]}
                      placeholder="Enter email"
                      type="email"
                    />
                  </div>,
                  <div
                    className="form-group"
                  >
                    <label>
                      Password
                    </label>
                    <input
                      className="form-control"
                      id="exampleInputPassword1"
                      name="password"
                      onChange={[Function]}
                      placeholder="Password"
                      type="password"
                    />
                  </div>,
                  <button
                    className="btn btn-primary"
                    type="submit"
                  >
                    Submit
                  </button>,
                ],
                "className": "loginform",
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <label>
                        Email Address
                      </label>,
                      <input
                        aria-describedby="emailHelp"
                        className="form-control"
                        id="email"
                        name="email"
                        onChange={[Function]}
                        placeholder="Enter email"
                        type="email"
                      />,
                    ],
                    "className": "form-group",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Email Address",
                      },
                      "ref": null,
                      "rendered": "Email Address",
                      "type": "label",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "aria-describedby": "emailHelp",
                        "className": "form-control",
                        "id": "email",
                        "name": "email",
                        "onChange": [Function],
                        "placeholder": "Enter email",
                        "type": "email",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <label>
                        Password
                      </label>,
                      <input
                        className="form-control"
                        id="exampleInputPassword1"
                        name="password"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
                      />,
                    ],
                    "className": "form-group",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Password",
                      },
                      "ref": null,
                      "rendered": "Password",
                      "type": "label",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "form-control",
                        "id": "exampleInputPassword1",
                        "name": "password",
                        "onChange": [Function],
                        "placeholder": "Password",
                        "type": "password",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Submit",
                    "className": "btn btn-primary",
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": "Submit",
                  "type": "button",
                },
              ],
              "type": "form",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Don't have an account?",
                "href": "/signup",
                "id": "signUpLink",
              },
              "ref": null,
              "rendered": "Don't have an account?",
              "type": "a",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="centerLogin"
        >
          <img
            alt="JobSage Logo"
            className="img-fluid"
            src="logo.png"
          />
          <h1
            className="display-3"
            id="applicationTitle"
          >
            JobSage
          </h1>
          <div
            className="form"
          >
            <form
              className="loginform"
              onSubmit={[Function]}
            >
              <div
                className="form-group"
              >
                <label>
                  Email Address
                </label>
                <input
                  aria-describedby="emailHelp"
                  className="form-control"
                  id="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="Enter email"
                  type="email"
                />
              </div>
              <div
                className="form-group"
              >
                <label>
                  Password
                </label>
                <input
                  className="form-control"
                  id="exampleInputPassword1"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                />
              </div>
              <button
                className="btn btn-primary"
                type="submit"
              >
                Submit
              </button>
            </form>
            <a
              href="/signup"
              id="signUpLink"
            >
              Don't have an account?
            </a>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="JobSage Logo"
              className="img-fluid"
              src="logo.png"
            />,
            <h1
              className="display-3"
              id="applicationTitle"
            >
              JobSage
            </h1>,
            false,
            <div
              className="form"
            >
              <form
                className="loginform"
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <label>
                    Email Address
                  </label>
                  <input
                    aria-describedby="emailHelp"
                    className="form-control"
                    id="email"
                    name="email"
                    onChange={[Function]}
                    placeholder="Enter email"
                    type="email"
                  />
                </div>
                <div
                  className="form-group"
                >
                  <label>
                    Password
                  </label>
                  <input
                    className="form-control"
                    id="exampleInputPassword1"
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                  />
                </div>
                <button
                  className="btn btn-primary"
                  type="submit"
                >
                  Submit
                </button>
              </form>
              <a
                href="/signup"
                id="signUpLink"
              >
                Don't have an account?
              </a>
            </div>,
          ],
          "className": "centerLogin",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "JobSage Logo",
              "className": "img-fluid",
              "src": "logo.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "JobSage",
              "className": "display-3",
              "id": "applicationTitle",
            },
            "ref": null,
            "rendered": "JobSage",
            "type": "h1",
          },
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <form
                  className="loginform"
                  onSubmit={[Function]}
                >
                  <div
                    className="form-group"
                  >
                    <label>
                      Email Address
                    </label>
                    <input
                      aria-describedby="emailHelp"
                      className="form-control"
                      id="email"
                      name="email"
                      onChange={[Function]}
                      placeholder="Enter email"
                      type="email"
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <label>
                      Password
                    </label>
                    <input
                      className="form-control"
                      id="exampleInputPassword1"
                      name="password"
                      onChange={[Function]}
                      placeholder="Password"
                      type="password"
                    />
                  </div>
                  <button
                    className="btn btn-primary"
                    type="submit"
                  >
                    Submit
                  </button>
                </form>,
                <a
                  href="/signup"
                  id="signUpLink"
                >
                  Don't have an account?
                </a>,
              ],
              "className": "form",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="form-group"
                    >
                      <label>
                        Email Address
                      </label>
                      <input
                        aria-describedby="emailHelp"
                        className="form-control"
                        id="email"
                        name="email"
                        onChange={[Function]}
                        placeholder="Enter email"
                        type="email"
                      />
                    </div>,
                    <div
                      className="form-group"
                    >
                      <label>
                        Password
                      </label>
                      <input
                        className="form-control"
                        id="exampleInputPassword1"
                        name="password"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
                      />
                    </div>,
                    <button
                      className="btn btn-primary"
                      type="submit"
                    >
                      Submit
                    </button>,
                  ],
                  "className": "loginform",
                  "onSubmit": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <label>
                          Email Address
                        </label>,
                        <input
                          aria-describedby="emailHelp"
                          className="form-control"
                          id="email"
                          name="email"
                          onChange={[Function]}
                          placeholder="Enter email"
                          type="email"
                        />,
                      ],
                      "className": "form-group",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Email Address",
                        },
                        "ref": null,
                        "rendered": "Email Address",
                        "type": "label",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "aria-describedby": "emailHelp",
                          "className": "form-control",
                          "id": "email",
                          "name": "email",
                          "onChange": [Function],
                          "placeholder": "Enter email",
                          "type": "email",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <label>
                          Password
                        </label>,
                        <input
                          className="form-control"
                          id="exampleInputPassword1"
                          name="password"
                          onChange={[Function]}
                          placeholder="Password"
                          type="password"
                        />,
                      ],
                      "className": "form-group",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Password",
                        },
                        "ref": null,
                        "rendered": "Password",
                        "type": "label",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "form-control",
                          "id": "exampleInputPassword1",
                          "name": "password",
                          "onChange": [Function],
                          "placeholder": "Password",
                          "type": "password",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Submit",
                      "className": "btn btn-primary",
                      "type": "submit",
                    },
                    "ref": null,
                    "rendered": "Submit",
                    "type": "button",
                  },
                ],
                "type": "form",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Don't have an account?",
                  "href": "/signup",
                  "id": "signUpLink",
                },
                "ref": null,
                "rendered": "Don't have an account?",
                "type": "a",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
